<?xml version="1.0"?>
<launch>
  <param name="/use_sim_time" value="false" />
  
  <include file="$(find hdl_people_tracking)/launch/mini_display.launch"/>

  <!-- arguments -->
 <!-- arguments -->
  <arg name="nodelet_manager" default="velodyne_nodelet_manager" />
  <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch">
  </include>

  <node pkg="nodelet" type="nodelet" name="hdl_people_detection_nodelet" args="load hdl_people_tracking/HdlPeopleDetectionNodelet $(arg nodelet_manager)">
  	<param name="static_sensor" value="true"/>
    <param name="downsample_resolution" value="0.05"/>
    <!-- back subtraction params -->
    <param name="backsub_resolution" value="0.2"/>
    <param name="backsub_occupancy_thresh" value="2"/>
    <!-- clustering params -->
    <param name="cluster_min_pts" value="10"/>
    <param name="cluster_max_pts" value="2048"/>
    <param name="cluster_min_size_x" value="0.2"/>
    <param name="cluster_min_size_y" value="0.2"/>
    <param name="cluster_min_size_z" value="-0.5"/>
    <param name="cluster_max_size_x" value="1.0"/>
    <param name="cluster_max_size_y" value="1.0"/>
    <param name="cluster_max_size_z" value="2.0"/>
    <!-- classification params -->
    <param name="enable_classification" value="true"/>

    <remap from="/velodyne_points" to="/ransac"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="hdl_people_tracking_nodelet" args="load hdl_people_tracking/HdlPeopleTrackingNodelet $(arg nodelet_manager)">
    <!-- tracks with larger covariance matrix trace than this value will be removed -->
    <param name="remove_trace_thresh" value="1.0"/>
    <param name="human_radius" value="0.4"/>
    <param name="using_svm" value = "false"/>
  </node>
  <node pkg="hdl_people_tracking" type="receive_pcl.py" name="receive_pcl" output="screen">
  </node>
  <node pkg="rviz" type="rviz" name="mytest_rviz" args="-d /home/chen/mytracking_ws/src/85/hdl_people_tracking/include/hdl_people_tracking/rviz_cxy.rviz" />
  <node pkg="learning_PCL" type="ransac" name="ransac"/>
  <node pkg="reflection_trajectory" type="turtle_tf_listener"  name="listener" output="screen"/>
  <node pkg="tf" type="static_transform_publisher" name="velo_to_laser" args="0 0 0 0 0 0 /base_link /velodyne 12" output="screen"/>
  <node pkg = "hdl_people_tracking" type = "mini_control" name = "mini_control">
        <param name="vel_x_factor" value="1.0" />
        <param name="local_port" value="43894"/>
        <param name="remote_port" value="43893"/>
        <param name="remote_ip" value="192.168.1.120"/>
  </node>
  <node pkg="hdl_people_tracking" type="pos2vel" name="pos2vel">
  <param name = "postopic" value = "/mytarget_position"/>
  <param name = "veltopic" value = "/cmd_vel"/>
  <param name = "lostopic" value = "/lost_flag"/>
  <param name = "kp_theta" value = "2.0"/>
  <param name = "kp" value ="0.1"/>
  <param name = "max_vel" value = "2.0"/>
  <param name = "min_dis" value = "2.0"/>
  <param name = "max_theta" value = "0.523"/>
  <param name = "max_lost_time" value = "10"/>
  </node>
  
</launch>
